metadata/src/main/java/org/apache/kafka/image/ClusterDelta.java

```diff
@@ -56,6 +58,15 @@ public BrokerRegistration broker(int nodeId) {
         return image.broker(nodeId);
     }
 
+    public Set<Integer> liveZkBrokerIdChanges() {
+        return changedBrokers
+            .values()
+            .stream()
+            .filter(broker -> broker.isPresent() && broker.get().isMigratingZkBroker())
```

Comment:
We can do `filter(Optional::isPresent)` followed by `map(Optional::get)` to make this a little neater (after we drop java 8 support, we'll be able to use Stream.flatMap with Optional.stream)